src/app/settings/views/Configuration/GeneralForm/GeneralForm.tsx

```diff
@@ -111,6 +118,31 @@ const GeneralForm = (): JSX.Element => {
         type="text"
         wrapperClassName="u-sv2"
       />
+      <p>MAAS theme main colour</p>
+      <Row className="generalForm__radio-row">
+        {[
+          "Default",
+          "Bark",
+          "Sage",
+          "Olive",
+          "Viridian",
+          "Prussian green",
+          "Blue",
+          "Purple",
+          "Magenta",
+          "Red",
+        ].map((color) => (
+          <Col medium={1} size={1} small={2}>
+            <FormikField
+              color={color.toLowerCase().split(" ").join("_")}
```

Comment:
Although it can be tempting to create classNames dynamically as it reduces repetition it has a downside of not being easily searchable and can be brittle.

I'd suggest to use an array `{ value: string, label: string }[]` instead (e.g. `[{ value: 'default', label: 'Default' }]`.

Reply:
As long as you provide a strict type for the color prop, I'm not too fussed about leaving this as is.

Reply:
Changed this over to static generation. Agreed it would probably better for long term stability and maintainability. Also made sure the values were strictly typed as well (using ColorValues)

```tsx
<Row className="general-form__radio-row">
  {[
    { value: ColorValues.Default, label: "Default" },
    { value: ColorValues.Bark, label: "Bark" },
    { value: ColorValues.Sage, label: "Sage" },
    { value: ColorValues.Olive, label: "Olive" },
    { value: ColorValues.Viridian, label: "Viridian" },
    { value: ColorValues.PrussianGreen, label: "Prussian green" },
    { value: ColorValues.Blue, label: "Blue" },
    { value: ColorValues.Purple, label: "Purple" },
    { value: ColorValues.Magenta, label: "Magenta" },
    { value: ColorValues.Red, label: "Red" },
  ].map((color) => (
    <Col medium={1} size={1} small={2}>
      <FormikField
        color={color.value}
        component={ThemedRadioButton}
        label={color.label}
        name="theme"
      />
    </Col>
  ))}
</Row>
```